src/server/game/Entities/GameObject/GameObject.cpp

```diff
@@ -2529,3 +2573,110 @@ GameObjectModel* GameObject::CreateModel()
 {
     return GameObjectModel::Create(std::make_unique<GameObjectModelOwnerImpl>(this), sWorld->GetDataPath());
 }
+
+bool GameObject::IsAtInteractDistance(Player const* player, SpellInfo const* spell) const
+{
+    if (spell || (spell = GetSpellForLock(player)))
```

Comment:
hmm something doesn't seem right here
(spell = GetSpellForLock(player))
Did they mean to do
(spell == GetSpellForLock(player)

It's like that even on latest TC I wonder.. https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Entities/GameObject/GameObject.cpp#L2690